readthedocs/rtd_tests/tests/test_builds.py

```diff
@@ -54,6 +57,165 @@ def test_build(self):
+            enable_pdf_build=False,
+            enable_epub_build=False,
+            versions=[fixture()]
+        )
+        version = project.versions.all()[0]
+        build_env = LocalBuildEnvironment(
+            project=project,
+            version=version,
+            build={}
+        )
+        python_env = Virtualenv(version=version, build_env=build_env)
+        base_builder = MkdocsHTML(build_env, python_env)
+
+        def look_index_path(path):
+            if path.endswith('README.md'):
+                return True
+            elif path.endswith('index.md'):
+                return True
+            return False
+
+        with mock.patch('readthedocs.doc_builder.base.open', mock.mock_open()) as mock_open, fake_paths(look_index_path):
+            result = base_builder.create_index(extension='md', force_index=True)
+            self.assertEqual(len(mock_open.mock_calls), 0)
+            self.assertEqual(result, 'index')
+
+    def test_build_generate_index_file(self):
+        project = get(
+            Project,
+            slug='project-1',
+            documentation_type='sphinx',
+            conf_py_file='',
+            enable_pdf_build=False,
+            enable_epub_build=False,
+            versions=[fixture()]
+        )
+        version = project.versions.all()[0]
+        build_env = LocalBuildEnvironment(
+            project=project,
+            version=version,
+            build={}
+        )
+        python_env = Virtualenv(version=version, build_env=build_env)
+        base_builder = SphinxHTML(build_env, python_env)
+
+        def look_index_path(path):
+            if path.endswith('README.md'):
+                return False
+            elif path.endswith('index.md'):
+                return False
+            return False
+
+        with mock.patch('readthedocs.doc_builder.base.open', mock.mock_open()) as mock_open, fake_paths(look_index_path):
+            result = base_builder.create_index(extension='md')
+            mock_open.assert_called_once_with(
+                os.path.join(base_builder.docs_dir(), 'index.md'), 'w+'
+            )
+            self.assertEqual(result, 'index')
+
+    def test_get_readme_file(self):
+        project = get(
+            Project,
+            slug='project-1',
+            documentation_type='sphinx',
+            conf_py_file='',
+            enable_pdf_build=False,
+            enable_epub_build=False,
+            versions=[fixture()]
+        )
+        version = project.versions.all()[0]
+        build_env = LocalBuildEnvironment(
+            project=project,
+            version=version,
+            build={}
+        )
+        python_env = Virtualenv(version=version, build_env=build_env)
+        base_builder = SphinxHTML(build_env, python_env)
+
+        def look_index_path(path):
+            if path.endswith('README.md'):
+                return True
+            elif path.endswith('index.md'):
+                return False
+            return False
```

Comment:
This is repeated 4 times, I think it should be a util method at the top of the file?

Reply:
I will add this method to the mock path utils better :)